src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseSign1Message.cs

```diff
@@ -0,0 +1,283 @@
+        {
+            if (content == null)
+            {
+                throw new ArgumentNullException(nameof(content));
+            }
+
+            if (key == null)
+            {
+                throw new ArgumentNullException(nameof(key));
+            }
+
+            byte[] encodedProtectedHeader = CreateEncodedProtectedHeader(KeyType.ECDsa, hashAlgorithm);
+            byte[] toBeSigned = CreateToBeSigned(SigStructureCoxtextSign1, encodedProtectedHeader, content);
+            byte[] signature = SignWithECDsa(key, toBeSigned, hashAlgorithm);
+            return SignCore(encodedProtectedHeader, GetEmptyCborMap(), signature, content, isDetached);
+        }
+
+        [UnsupportedOSPlatform("browser")]
+        public static byte[] Sign(byte[] content, RSA key, HashAlgorithmName hashAlgorithm, bool isDetached = false)
+        {
+            if (content == null)
+            {
+                throw new ArgumentNullException(nameof(content));
+            }
+
+            if (key == null)
+            {
+                throw new ArgumentNullException(nameof(key));
+            }
+
+            byte[] encodedProtectedHeader = CreateEncodedProtectedHeader(KeyType.RSA, hashAlgorithm);
+            byte[] toBeSigned = CreateToBeSigned(SigStructureCoxtextSign1, encodedProtectedHeader, content);
+            byte[] signature = SignWithRSA(key, toBeSigned, hashAlgorithm);
+            return SignCore(encodedProtectedHeader, GetEmptyCborMap(), signature, content, isDetached);
+        }
+
+        [UnsupportedOSPlatform("browser")]
+        public static byte[] Sign(byte[] content, CoseHeaderMap protectedHeaders, CoseHeaderMap unprotectedHeaders, AsymmetricAlgorithm key, bool isDetached = false)
+        {
+            if (content == null)
+            {
+                throw new ArgumentNullException(nameof(content));
+            }
+
+            if (key == null)
+            {
+                throw new ArgumentNullException(nameof(key));
+            }
+
+            if (protectedHeaders == null)
+            {
+                throw new ArgumentNullException(nameof(protectedHeaders));
+            }
+
+            if (unprotectedHeaders == null)
+            {
+                throw new ArgumentNullException(nameof(unprotectedHeaders));
+            }
+
+            KeyType keyType = key switch
+            {
+                ECDsa => KeyType.ECDsa,
+                RSA => KeyType.RSA,
+                _ => throw new CryptographicException(SR.Format(SR.Sign1SignUnsupportedKey, key.GetType()))
+            };
+
+            ThrowIfDuplicateLabels(protectedHeaders, unprotectedHeaders);
+
+            ReadOnlyMemory<byte> encodedAlg = GetCoseAlgorithmFromBuckets(protectedHeaders, unprotectedHeaders);
+
+            int? algorithmHeader = DecodeCoseAlgorithmHeader(encodedAlg);
+            Debug.Assert(algorithmHeader.HasValue, "Algorithm (alg) is a known header and should have been validated in Set[Encoded]Value()");
```

Comment:
I thought that we were going to validate it if specified, and write it if not?  This looks like it requires the caller to specify the algorithm header instead of preferring that they didn't.

Reply:
I thought we were going to do that only for the "easy-to-use" overloads since that is one less assumption. 
Also it is non-breaking to go from exception to enable but it is breaking vice-versa.